_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -169,57 +310,14 @@ mfxStatus MFXVideoENCODE_Query(mfxSession session, mfxVideoParam *in, mfxVideoPa
         else
 #endif
         {
-            switch (out->mfx.CodecId)
-            {
-#ifdef MFX_ENABLE_H264_VIDEO_ENCODE
-            case MFX_CODEC_AVC:
-#if defined (MFX_ENABLE_H264_VIDEO_ENCODE_HW)
-                if (!session->m_pENCODE.get())
-                    mfxRes = MFXHWVideoENCODEH264::Query(session->m_pCORE.get(), in, out);
-                else
-                    mfxRes = MFXHWVideoENCODEH264::Query(session->m_pCORE.get(), in, out, session->m_pENCODE.get());
-#else //MFX_VA
-                mfxRes = MFXVideoENCODEH264::Query(in, out);
-#endif //MFX_VA
-                break;
-#endif
-
-#ifdef MFX_ENABLE_MPEG2_VIDEO_ENCODE
-            case MFX_CODEC_MPEG2:
-                mfxRes = MFXVideoENCODEMPEG2_HW::Query(session->m_pCORE.get(), in, out);
-                break;
-#endif
-
-#if defined(MFX_ENABLE_MJPEG_VIDEO_ENCODE)
-            case MFX_CODEC_JPEG:
-                mfxRes = MFXVideoENCODEMJPEG_HW::Query(session->m_pCORE.get(), in, out);
-                break;
-#endif // MFX_ENABLE_MJPEG_VIDEO_ENCODE
-
-#if defined(MFX_ENABLE_H265_VIDEO_ENCODE)
-            case MFX_CODEC_HEVC:
-            {
-#if defined (MFX_ENABLE_HEVC_VIDEO_FEI_ENCODE)
-                bool * feiEnabled = (bool*)session->m_pCORE->QueryCoreInterface(MFXIFEIEnabled_GUID);//required to check FEI plugin registration.
-                MFX_CHECK_NULL_PTR1(feiEnabled);
-                if(*feiEnabled)
-                    mfxRes = MfxHwH265FeiEncode::H265FeiEncode_HW::Query(session->m_pCORE.get(), in, out);
-                else
-#endif
-                    mfxRes = MfxHwH265Encode::MFXVideoENCODEH265_HW::Query(session->m_pCORE.get(), in, out);
-                break;
-            }
-#endif // MFX_ENABLE_H265_VIDEO_ENCODE
+            // required to check FEI plugin registration
+            bool *feiEnabled = (bool*)session->m_pCORE->QueryCoreInterface(MFXIFEIEnabled_GUID);
+            MFX_CHECK_NULL_PTR1(feiEnabled);
+            const bool fei = *feiEnabled;
```

Comment:
But I like the idea with check_fei inline helper. It will simplify this and few other places to just:
`const bool fei = check_fei(session, codec)`

Reply:
Unfortunately with error processing it may not fit in single line.

Reply:
@Alexandr-Konovalov,

Do u mean that we need to return [MFX_ERR_NULL_PTR] if `QueryCoreInterface(MFXIFEIEnabled_GUID)` returns NULL?

Reply:
Yes, for queries. And nullptr from Create().